s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/output/ByteArrayWriter.java

```diff
@@ -35,18 +37,33 @@ public void configureValueConverter(final Map<String, String> config, final S3So
 
     }
 
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     @Override
-    public List<Object> getRecords(final InputStream inputStream, final String topic, final int topicPartition) {
-        return List.of(inputStream);
-    }
+    public List<Object> getRecords(final InputStream inputStream, final String topic, final int topicPartition,
+            final S3SourceConfig s3SourceConfig) {
 
-    @Override
-    public byte[] getValueBytes(final Object record, final String topic, final S3SourceConfig s3SourceConfig) {
+        final int maxMessageBytesSize = s3SourceConfig.getInt(MAX_MESSAGE_BYTES_SIZE);
+        final byte[] buffer = new byte[maxMessageBytesSize];
+        int bytesRead;
+
+        final List<Object> chunks = new ArrayList<>();
         try {
-            return IOUtils.toByteArray((InputStream) record);
+            bytesRead = inputStream.read(buffer);
+            while (bytesRead != -1) {
+                final byte[] chunk = new byte[bytesRead];
+                System.arraycopy(buffer, 0, chunk, 0, bytesRead);
+                chunks.add(chunk);
+                bytesRead = inputStream.read(buffer);
+            }
         } catch (IOException e) {
-            LOGGER.error("Error in reading s3 object stream " + e.getMessage());
-            return new byte[0];
+            LOGGER.error("Error reading from input stream: " + e.getMessage(), e);
```

Comment:
We should be using the format
LOGGER.error("Error reading from input stream:  {}", e.getMessage(), e);